gpcontrib/arenadata_toolkit/arenadata_toolkit--1.0--1.1.sql

```diff
@@ -1,3 +1,50 @@
 /* gpcontrib/arenadata_toolkit/arenadata_toolkit--1.0--1.1.sql */
 
 ALTER FUNCTION arenadata_toolkit.adb_get_relfilenodes(tablespace_oid OID) ROWS 30000000;
+
+CREATE OR REPLACE FUNCTION arenadata_toolkit.adb_relation_storage_size_on_segments(
+	IN reloid OID, IN forkName TEXT default 'main',
+	OUT gp_segment_id INT, OUT size BIGINT)
+AS 'SELECT pg_catalog.gp_execution_segment() AS gp_segment_id, *
+	FROM arenadata_toolkit.adb_relation_storage_size($1, $2)'
+LANGUAGE SQL EXECUTE ON ALL SEGMENTS;
+
+GRANT EXECUTE ON FUNCTION arenadata_toolkit.adb_relation_storage_size_on_segments(OID, TEXT) TO public;
+
+CREATE OR REPLACE VIEW arenadata_toolkit.adb_skew_coefficients
+AS
+WITH recursive cte AS (
+	SELECT
+		oid AS id,
+		tables_size.gp_segment_id AS seg_id,
+		tables_size.size AS size
+	FROM pg_catalog.pg_class,
+		 arenadata_toolkit.adb_relation_storage_size_on_segments(oid) AS tables_size
+	WHERE relkind = 'r' AND relstorage != 'x' AND
+		relnamespace IN (SELECT aunoid FROM gp_toolkit.__gp_user_namespaces)
+	UNION ALL
+		SELECT inhparent AS id, seg_id, size
+		FROM cte
+		LEFT JOIN pg_inherits ON inhrelid = id
+		WHERE inhparent != 0
+), tables_size_by_segments AS (
+	SELECT id, sum(size) AS size
+	FROM cte
+	GROUP BY id, seg_id
+), skew AS (
+	SELECT
+		id AS skewoid,
+		stddev(size) AS skewdev,
+		avg(size) AS skewmean
+	FROM tables_size_by_segments
+	GROUP BY id)
+SELECT
+	skew.skewoid AS skcoid,
+	pgn.nspname  AS skcnamespace,
+	pgc.relname  AS skcrelname,
+	CASE WHEN skewdev > 0 THEN skewdev/skewmean * 100.0 ELSE 0 END AS skccoeff
```

Comment:
Why is the numerator checked for zero, and not the denominator? It is clear that if the numerator is not zero, then neither is the denominator, but still...

Reply:
Because if table has same sizes at all segments (for example 32768) than skewmean is 32768 and skewdev is 0. You can do `skewdev/skewmean * 100.0` but you will get 0. Why do we have to do extra calculations (It will be if change `skewdev > 0` to `skewmean > 0`)?